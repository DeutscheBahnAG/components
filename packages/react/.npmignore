# Documentation
*-doc.js
*-doc.scss
*-doc.css
# Tests
*.snap
*.test.*
#
.release-it.js
tsconfig.build.json
wss_config.yml
