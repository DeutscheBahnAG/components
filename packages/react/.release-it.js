module.exports = {
  git: {
    commitMessage: 'chore: Release @bahn-x/react@${version}',
    requireBranch: 'main',
    tagName: '@bahn-x/react@${version}',
  },
  npm: {
    publishPath: 'dist',
  },
};
