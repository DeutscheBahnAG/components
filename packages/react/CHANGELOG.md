# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

Make sure to format this file using Prettier.

## [Unreleased]

## [1.1.1] - 2022-06-28

### Fixed

- Require latest design fixes of `@db-design/styles`

## [1.1.0] - 2022-06-28

### Added

- Columns support for Container

## [1.0.1] - 2022-05-11

### Fixed

- Missing inline spacer for `<Track>`

## [1.0.0] - 2022-05-10

### Changed

- Renamed `mt`/`mb` to `logoUnitsTop`/`logoUnitsBottom` for `<Logo>` and `<Pulse>`

## [1.0.0-pre.24] - 2022-03-23
