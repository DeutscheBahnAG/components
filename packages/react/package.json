{
  "name": "@db-design/react",
  "version": "1.0.0-pre.19",
  "description": "DB React Components",
  "repository": {
    "directory": "packages/react",
    "url": "git@ssh.git.tech.rz.db.de:bahn-x/webdev/web.git",
    "type": "git"
  },
  "license": "SEE LICENSE IN LICENSE.md",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "dependencies": {
    "@db-design/react-icons": "^1.0.0-pre.1",
    "@db-design/styles": "^1.0.0-pre.17",
    "clsx": "^1.1.1",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-modal": "3.12.1"
  },
  "devDependencies": {
    "@testing-library/react": "^12.1.2",
    "@types/react-modal": "^3.10.6",
    "release-it": "^14.2.2",
    "typescript": "^4.1.3"
  },
  "scripts": {
    "build": "rm -rf dist && tsc --project tsconfig.build.json",
    "postbuild": "./postbuild.sh",
    "release": "release-it --disable-metrics"
  },
  "sideEffects": [
    "./dist/button/*",
    "./dist/modal/*",
    "./dist/pulse/*",
    "./dist/triptime/*",
    "./dist/triptimespan/*"
  ]
}
