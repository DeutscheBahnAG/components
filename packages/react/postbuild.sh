#! /bin/bash

cp -p package.json dist &&
for f in `ls *.md`; do cp -p $f dist; done &&
for f in `ls src/*.d.ts`; do cp -p $f dist; done &&
for f in `ls src/*/README.md`; do cp -p $f ${f/src/dist}; done &&
perl -i -pe 's#dist/index#index#' dist/package.json
perl -i -pe 's#dist/module/index#module/index#' dist/package.json
