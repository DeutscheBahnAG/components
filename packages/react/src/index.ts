export * as TransportchipIcons from './transportchip/icons';

export { default as Anchor } from './anchor';
export { default as Button } from './button';
export { default as Checkbox } from './checkbox';
export { default as Container } from './container';
export { default as Copy } from './copy';
export { default as Layout } from './layout';
export { default as Link } from './link';
export { default as Loadingindicator } from './loadingindicator';
export { default as Logo } from './logo';
export { default as Modal } from './modal';
export { default as Notification } from './notification';
export { default as Passwordfield } from './passwordfield';
export { default as Pulse } from './pulse';
export { default as Select } from './select';
export { default as Status } from './status';
export { default as Textfield } from './textfield';
export { default as Theme } from './theme';
export { default as Time } from './time';
export { default as Title } from './title';
export { default as Track } from './track';
export { default as Transportchip } from './transportchip';
export { default as Triptime } from './triptime';
export { default as Triptimespan } from './triptimespan';
export { default as Radiobutton } from './radiobutton';
export { default as Accordion } from './accordion';
export { default as Accordionitem } from './accordionitem';
export { default as List } from './list';
