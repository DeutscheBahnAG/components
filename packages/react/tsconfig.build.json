{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "esModuleInterop": true,
    "jsx": "react",
    "lib": ["dom", "ES2020"],
    "noImplicitAny": false,
    "outDir": "dist",
    "target": "ES2015",
    "sourceMap": false
  },
  "include": ["src"],
  "exclude": ["**/*.test.*", "**/*-doc.*", "**/*.snap"]
}
