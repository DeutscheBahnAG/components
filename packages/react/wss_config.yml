#DISCLAIMER
#All sensible data is not persisted and deleted shortly after usage. So be assured we cope according to any law with your data!

# NOTE: see here how these configs are used https://git.tech.rz.db.de/ass/dependencychecker/whitesource/whitesource-dockered-scanner/-/blob/master/Docker/base/wrapper/scan

##################
#   WHITESOURCE  #
##################
wss_project_name="db-design-react"
wss_project_version="$CI_COMMIT_BRANCH"
  #wss_product_version="latest" DO NOT SET THIS IF USING wss_project_version as this overrides it

  # The following variables must be set as env variables in the Gitlab bahn-x group, they are valid for all subgroups
  #
  # wss_apiKey  -> your organization key, from Whitesource WebGUI:  Integrate > Organization > API key
  # wss_userKey -> CI tech-user key, as received initially via email, also found in 1pass

  # The following variables must be set as env variables in the Gitlab bahn-x subgroups like TMP, Analytics etc
  #
  # wss_product_name  -> set set this to one of our whitesource "products"  https://whitesource.int-app.comp.db.de/Wss/WSS.html#!allProducts
  # wss_product_token -> the products key, from Whitesource WebGUI:  Integrate > Product Tokens > Token

  # Npm specific config: If you have scoped dependencies, please add them according to the conventions below:
  # npmScopedDeps="@yourScope:registry=yourRegistryLink;@yourScope2:registry=yourRegistryLink2"
npmScopedDeps="@bahn-x:registry=https://npm.pkg.github.com"

  # Whitesource will scan your entire project by deafult.
  # Change this, if you want scan only a specific path.
wss_scan_folder=packages/react/

# You feel brave today?
# Great, use your own wss-unified-agent.config file as of here: https://whitesource.atlassian.net/wiki/spaces/WD/pages/489160834/Unified+Agent+Configuration+File+Parameters
# Tell us, where we will find it in your repo structure, starting from the root directory.
# no leading slash!!
#
# !! Please note that in this case all information (including project / product name and version) must be specified in this team-specific Whitesource config file and will no longer be read from wss_config.yml
# (apart from wss_scan_folder and wss_javaParams which is meta-information (i.e. not Whitesource specific) and therefore still needs to be specified).
#wss_specific_team_config=whitesource/wss-unified-agent.config
