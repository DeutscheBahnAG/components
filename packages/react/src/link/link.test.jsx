import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ActionAdd } from '@db-design/react-icons';
import Link from './link';

describe('Link Component', () => {
  it('should render a link', () => {
    const { container } = render(<Link href="/">My link 1</Link>);
    const link = screen.getByRole('link');
    expect(link).toHaveTextContent('My link 1');
    expect(container).toMatchSnapshot();
  });

  it('should render a submit link', () => {
    const { container } = render(<Link type={Link.types.SUBMIT}>My link 2</Link>);
    expect(container).toMatchSnapshot();
  });

  it('should render a secondary link', () => {
    const { container } = render(<Link variant={Link.variants.SECONDARY}>My link 3</Link>);
    expect(container).toMatchSnapshot();
  });

  it('should render a mixed link', () => {
    const { container } = render(<Link variant={Link.variants.MIXED}>My link 4</Link>);
    expect(container).toMatchSnapshot();
  });

  it('should render a link with additional attributes', () => {
    const { container } = render(<Link title="Click me">My link 5</Link>);
    expect(container).toMatchSnapshot();
  });

  it('should render a link with additional classes', () => {
    const { container } = render(<Link className="my-class">My link 6</Link>);
    expect(container).toMatchSnapshot();
  });

  it('should render a link when `href` is not set', () => {
    const { container } = render(<Link>My link 7</Link>);
    expect(container).toMatchSnapshot();
  });

  it('should render an internal link anchor when `href` is set', () => {
    const { container } = render(<Link href="/">My link 8</Link>);
    expect(container).toMatchSnapshot();
  });

  it('should render an download link anchor when `href` is set', () => {
    const { container } = render(
      <Link href="/" download>
        My link 9
      </Link>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render an external link anchor when `href` starts with http', () => {
    const { container } = render(<Link href="http://www.bahn.de/">My link 10</Link>);
    expect(container).toMatchSnapshot();
  });

  it('should render an external link anchor when `href` starts with https', () => {
    const { container } = render(<Link href="https://www.bahn.de/">My link 11</Link>);
    expect(container).toMatchSnapshot();
  });

  it('should render a the icon on the left side', () => {
    const { container } = render(
      <Link href="/" iconPosition={Link.iconPositions.BEFORE}>
        My link 12
      </Link>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render a the icon on the right side', () => {
    const { container } = render(
      <Link href="/" iconPosition={Link.iconPositions.AFTER} download>
        My link 13
      </Link>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render a custom icon', () => {
    const { container } = render(
      <Link href="/" icon={<ActionAdd />}>
        My link 14
      </Link>
    );
    expect(container).toMatchSnapshot();
  });

  it('should allow custom class names', () => {
    const { container } = render(<Link className="my-custom-class">My link 15</Link>);
    expect(container).toMatchSnapshot();
  });

  it('should render a link if both href and onClick are defined', () => {
    const { container } = render(
      <Link href="/" onClick={() => {}}>
        My link 16
      </Link>
    );
    expect(container).toMatchSnapshot();
  });
});
