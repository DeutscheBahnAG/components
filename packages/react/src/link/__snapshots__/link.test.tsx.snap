// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component should allow custom class names 1`] = `
<div>
  <button
    class="db-link db-link--primary db-link--icon my-custom-class"
    data-testid="button"
    type="button"
  >
    <span>
      My link 15
    </span>
  </button>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Link Component should render a custom icon 1`] = `
<div>
  <a
    class="db-link db-link--primary db-link--icon"
    data-testid="link"
    href="/"
  >
    <svg
      class="db-icon"
      height="24"
      role="img"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 5c.564 0 1.01.446 1.01 1.01v4.98h4.98c.564 0 1.01.446 1.01 1.01 0 .564-.446 1.01-1.01 1.01h-4.98v4.98c0 .564-.446 1.01-1.01 1.01-.564 0-1.01-.446-1.01-1.01v-4.98H6.01C5.446 13.01 5 12.564 5 12c0-.564.446-1.01 1.01-1.01h4.98V6.01c0-.564.446-1.01 1.01-1.01z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
    <span>
      My link 14
    </span>
  </a>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Link Component should render a link 1`] = `
<div>
  <a
    class="db-link db-link--primary db-link--icon"
    data-testid="link"
    href="/"
  >
    <span>
      My link 1
    </span>
    <svg
      class="db-link__icon-arrow"
      height="24"
      role="img"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 12c0 .563.438 1 1 1h11.594l-4.282 4.281A.998.998 0 0012 18c0 .438.375 1 1.031 1 .25 0 .5-.094.688-.281l6-6A1 1 0 0020 12c.031-.25-.078-.515-.281-.719l-6-6A1.026 1.026 0 0013.03 5C12.47 5 12 5.469 12 6c0 .281.109.515.313.719L16.593 11H5c-.563 0-1 .438-1 1z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </a>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Link Component should render a link if both href and onClick are defined 1`] = `
<div>
  <a
    class="db-link db-link--primary db-link--icon"
    data-testid="link"
    href="/"
  >
    <span>
      My link 16
    </span>
    <svg
      class="db-link__icon-arrow"
      height="24"
      role="img"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 12c0 .563.438 1 1 1h11.594l-4.282 4.281A.998.998 0 0012 18c0 .438.375 1 1.031 1 .25 0 .5-.094.688-.281l6-6A1 1 0 0020 12c.031-.25-.078-.515-.281-.719l-6-6A1.026 1.026 0 0013.03 5C12.47 5 12 5.469 12 6c0 .281.109.515.313.719L16.593 11H5c-.563 0-1 .438-1 1z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </a>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Link Component should render a link when \`href\` is not set 1`] = `
<div>
  <button
    class="db-link db-link--primary db-link--icon"
    data-testid="button"
    type="button"
  >
    <span>
      My link 7
    </span>
  </button>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Link Component should render a link with additional attributes 1`] = `
<div>
  <button
    class="db-link db-link--primary db-link--icon"
    data-testid="button"
    title="Click me"
    type="button"
  >
    <span>
      My link 5
    </span>
  </button>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Link Component should render a link with additional classes 1`] = `
<div>
  <button
    class="db-link db-link--primary db-link--icon my-class"
    data-testid="button"
    type="button"
  >
    <span>
      My link 6
    </span>
  </button>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Link Component should render a mixed link 1`] = `
<div>
  <button
    class="db-link db-link--mixed db-link--icon"
    data-testid="button"
    type="button"
  >
    <span>
      My link 4
    </span>
  </button>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Link Component should render a secondary link 1`] = `
<div>
  <button
    class="db-link db-link--secondary db-link--icon"
    data-testid="button"
    type="button"
  >
    <span>
      My link 3
    </span>
  </button>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Link Component should render a submit link 1`] = `
<div>
  <button
    class="db-link db-link--primary db-link--icon"
    data-testid="button"
    type="submit"
  >
    <span>
      My link 2
    </span>
  </button>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Link Component should render a the icon on the left side 1`] = `
<div>
  <a
    class="db-link db-link--primary db-link--icon"
    data-testid="link"
    href="/"
  >
    <svg
      class="db-link__icon-arrow"
      height="24"
      role="img"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20.005 12c0-.563-.437-1-1-1H7.412l4.28-4.281A.998.998 0 0012.006 6c0-.438-.375-1-1.03-1a.97.97 0 00-.688.281l-6 6a.975.975 0 00-.277.611L4.005 12c-.03.25.078.515.282.719l6 6c.187.187.468.281.687.281.563 0 1.031-.469 1.031-1a.981.981 0 00-.312-.719L7.412 13h11.593c.563 0 1-.438 1-1z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
    <span>
      My link 12
    </span>
  </a>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Link Component should render a the icon on the right side 1`] = `
<div>
  <a
    class="db-link db-link--primary db-link--icon"
    data-testid="link"
    href="/"
  >
    <span>
      My link 13
    </span>
    <svg
      class="db-icon"
      height="24"
      role="img"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 22h12a3 3 0 003-3v-4c0-.563-.438-1-1-1-.563 0-1 .438-1 1v4c0 .563-.438 1-1 1H6c-.563 0-1-.438-1-1v-4c0-.563-.438-1-1-1-.563 0-1 .438-1 1v4a3 3 0 003 3zm6-6c.25 0 .531-.093.719-.28l3.969-3.992a.902.902 0 00.312-.717c0-.53-.438-.966-1-.966-.25 0-.531.093-.719.28L13 12.601V2.998C13 2.437 12.531 2 12 2c-.563 0-1 .437-1 .998V12.6l-2.281-2.276a1.102 1.102 0 00-.719-.28c-.563 0-1 .436-1 .997 0 .25.094.5.313.686l3.968 3.991c.188.188.469.281.719.281z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </a>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Link Component should render an download link anchor when \`href\` is set 1`] = `
<div>
  <a
    class="db-link db-link--primary db-link--icon"
    data-testid="link"
    href="/"
  >
    <svg
      class="db-icon"
      height="24"
      role="img"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 22h12a3 3 0 003-3v-4c0-.563-.438-1-1-1-.563 0-1 .438-1 1v4c0 .563-.438 1-1 1H6c-.563 0-1-.438-1-1v-4c0-.563-.438-1-1-1-.563 0-1 .438-1 1v4a3 3 0 003 3zm6-6c.25 0 .531-.093.719-.28l3.969-3.992a.902.902 0 00.312-.717c0-.53-.438-.966-1-.966-.25 0-.531.093-.719.28L13 12.601V2.998C13 2.437 12.531 2 12 2c-.563 0-1 .437-1 .998V12.6l-2.281-2.276a1.102 1.102 0 00-.719-.28c-.563 0-1 .436-1 .997 0 .25.094.5.313.686l3.968 3.991c.188.188.469.281.719.281z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
    <span>
      My link 9
    </span>
  </a>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Link Component should render an external link anchor when \`href\` starts with http 1`] = `
<div>
  <a
    class="db-link db-link--primary db-link--icon"
    data-testid="link"
    href="http://www.bahn.de/"
  >
    <span>
      My link 10
    </span>
    <svg
      class="db-link__icon-arrow"
      height="24"
      role="img"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.031 18a.953.953 0 00.657-.281L16 9.406V15c0 .594.469 1.031 1 1.031s1-.468 1-1.031V6.906C18 6.375 17.531 6 17 6H9c-.594 0-1 .469-1 1 0 .5.469 1 1 1h5.594l-8.375 8.375c-.157.156-.219.406-.219.594C6 17.562 6.438 18 7.031 18z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </a>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Link Component should render an external link anchor when \`href\` starts with https 1`] = `
<div>
  <a
    class="db-link db-link--primary db-link--icon"
    data-testid="link"
    href="https://www.bahn.de/"
  >
    <span>
      My link 11
    </span>
    <svg
      class="db-link__icon-arrow"
      height="24"
      role="img"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.031 18a.953.953 0 00.657-.281L16 9.406V15c0 .594.469 1.031 1 1.031s1-.468 1-1.031V6.906C18 6.375 17.531 6 17 6H9c-.594 0-1 .469-1 1 0 .5.469 1 1 1h5.594l-8.375 8.375c-.157.156-.219.406-.219.594C6 17.562 6.438 18 7.031 18z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </a>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Link Component should render an internal link anchor when \`href\` is set 1`] = `
<div>
  <a
    class="db-link db-link--primary db-link--icon"
    data-testid="link"
    href="/"
  >
    <span>
      My link 8
    </span>
    <svg
      class="db-link__icon-arrow"
      height="24"
      role="img"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 12c0 .563.438 1 1 1h11.594l-4.282 4.281A.998.998 0 0012 18c0 .438.375 1 1.031 1 .25 0 .5-.094.688-.281l6-6A1 1 0 0020 12c.031-.25-.078-.515-.281-.719l-6-6A1.026 1.026 0 0013.03 5C12.47 5 12 5.469 12 6c0 .281.109.515.313.719L16.593 11H5c-.563 0-1 .438-1 1z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </a>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;
